### PR TITLE
Bump kubectl to match the k8s deps in go.mod

### PR DESCRIPTION
**Pull request by Sodman**
# Description

 Bump kubectl to match the k8s deps in go.mod
 
 Addresses CVE-2023-45288 caught by Snyk scanner (golang http2 lib, Rated HIGH)

## API changes

N/A
## Code changes

N/A

## CI changes

N/A - CI is already testing k8s 1.31 in KinD

## Docs changes

N/A

# Context

CVE was caught by regular daily Snyk Scanning

This image is also used in Gloo Platform ([tracking issue](https://github.com/solo-io/gloo-mesh-enterprise/issues/19119))

## Testing steps

Re-ran existing tests, which make use of this image.

## Notes for reviewers

# Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by andy-fong — APPROVED** [review]

File: jobs/kubectl/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ARG BASE_IMAGE
 
-FROM bitnami/kubectl:1.29.6 as kubectl
+FROM bitnami/kubectl:1.31.1 as kubectl
```
Comment (jenshu) [inline jobs/kubectl/Dockerfile:3]: can you also modify `jobs/kubectl/Dockerfile.distroless`?

Comment (Sodman) [reply jobs/kubectl/Dockerfile:3]: Great catch! Done

File: changelog/v1.18.0-beta35/bump-kubectl-image.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+changelog:
+  - type: NON_USER_FACING
```
Comment (jenshu) [inline changelog/v1.18.0-beta35/bump-kubectl-image.yaml:2]: nit: should this be user-facing? maybe FIX

**Review by jenshu — APPROVED** [review]

File: changelog/v1.18.0-beta35/bump-kubectl-image.yaml (hunk 2 above)
Comment (Sodman) [reply changelog/v1.18.0-beta35/bump-kubectl-image.yaml:2]: I wasn't sure if this was actually a user facing image or just something used in our tests - I couldn't find references anywhere to it in this repo outside of tests.

In GME this image is conditionally used for an upgrade job related to automating migration of breaking prometheus changes, so I'll have a user-facing changelog there when I bump it in that repo.

Comment (jenshu) [reply changelog/v1.18.0-beta35/bump-kubectl-image.yaml:2]: we actually use this image in some jobs run during install/upgrade - see [rollout job](https://github.com/solo-io/gloo/blob/main/install/helm/gloo/templates/5-resource-rollout-job.yaml) and where it gets its [image value](https://github.com/solo-io/gloo/blob/833eb9d7d1814630f3d11ff2173a1028de829529/install/helm/gloo/values-template.yaml#L171) from (so it's not used for testing only)

Comment (Sodman) [reply changelog/v1.18.0-beta35/bump-kubectl-image.yaml:2]: Great, I'm sold, will change to `FIX` then so it's user facing.

Comment (Sodman) [inline changelog/v1.18.0-beta35/bump-kubectl-image.yaml:2]:
```suggestion
  - type: FIX
```

**Review by jenshu — APPROVED** [review]

**Pull request merged**